modules/rosetta/src/test/scala/org/tessellation/rosetta/domain/construction/ConstructionServiceSuite.scala

```diff
@@ -44,4 +59,51 @@ object ConstructionServiceSuite extends MutableIOSuite with Checkers {
       expect.eql(Left(InvalidPublicKey), _)
     }
   }
+
+  test("Successfully deserialize a transaction from hex") { res =>
+    implicit val securityProvider = res._1
+    implicit val kryo = res._2
+
+    val cs = ConstructionService.make[IO]()
+
+    KeyPairGenerator.makeKeyPair[IO].flatMap { srcKey =>
+      KeyPairGenerator.makeKeyPair[IO].flatMap { dstKey =>
+        val srcAddress = srcKey.getPublic.toAddress
+        val dstAddress = dstKey.getPublic.toAddress
+
+        generateTransactions(srcAddress, srcKey, dstAddress, 1).flatMap(_.map { hashedTransaction =>
+          val signedTransaction = hashedTransaction.signed
+
+          KryoSerializer[F]
+            .serialize(signedTransaction)
+            .map { bytes =>
+              val hex = Hex.fromBytes(bytes)
+
+              cs.deserializeTransactionFromHex(hex)
+                .rethrowT
+                .map(
+                  expect.eql(hashedTransaction.hash, _)
+                )
+            }
+            .toOption
+            .getOrElse(IO.pure(expect(false)))
+        }.head)
```

Comment:
if you want to check a single transaction, you can link there a valid hex for simplicity and that's all.
when you use a tx generator, and you already use a non empty list for transactions, you can create multiple txs and chain expectations:
```scala
    implicit val (s, k) = res

    val cs = ConstructionService.make[IO]()

    KeyPairGenerator.makeKeyPair[IO].flatMap { srcKey =>
      KeyPairGenerator.makeKeyPair[IO].flatMap { dstKey =>
        val srcAddress = srcKey.getPublic.toAddress
        val dstAddress = dstKey.getPublic.toAddress
        val txCount = PosInt(100)

        generateTransactions(srcAddress, srcKey, dstAddress, txCount)
          .flatMap(_.traverse { hashedTransaction =>
            KryoSerializer[F]
              .serialize(hashedTransaction.signed)
              .map(Hex.fromBytes(_))
              .map((_, hashedTransaction.hash))
              .liftTo[IO]
          })
          .flatMap {
            _.traverse {
              case (hex, hash) =>
                cs.deserializeTransactionFromHex(hex)
                  .rethrowT
                  .map(expect.eql(hash, _))
            }
          }
          .map(_.fold)
      }
    }
  }

```

Reply:
Changed.